src/main/java/net/minecraftforge/common/crafting/CraftingHelper.java

```diff
@@ -582,20 +584,17 @@ public static void loadRecipes()
         //TODO: Figure out how to remove recipes, and override them. This relies on cpw to help.
         //For now this is only done one after mod init, I want to move this to ServerInit and re-do it many times.
         init();
+        ForgeRegistry<IRecipe> reg = (ForgeRegistry<IRecipe>)ForgeRegistries.RECIPES;
         if (DEBUG_LOAD_MINECRAFT)
-        {
-            Iterator<IRecipe> itr = GameData.getRecipeRegistry().iterator();
-            while(itr.hasNext())
-            {
-                itr.next();
-                itr.remove();
-            }
-        }
+            reg.clear();
+        else
+            GameData.revert(RegistryManager.FROZEN, GameData.RECIPES, false);
```

Comment:
Recipes will be loaded from mod jars IDEALLY once. And then frozen like everything else in the registry system. This would gather 99% of recipes.
Then on server start it will load custom recipes from the server/world. This should be a far smaller number. 
Either way the actual recipe loading system should be a fairly trivial amount of that startup time. 
The fact that you say WTF means you haven't done any research into why it does what it does. 
Everything in the MC loading cycle is done for a reason.